.travis.yml

```diff
@@ -69,37 +60,14 @@ install:
-- if [ "$RUN_E2E_TESTS_EXPLORATION_STATISTICS_TAB" == 'true' ]; then travis_retry python -m scripts.run_e2e_tests --suite="explorationStatisticsTab" --prod_env; fi
-- if [ "$RUN_E2E_TESTS_EXPLORATION_TRANSLATION_TAB" == 'true' ]; then travis_retry python -m scripts.run_e2e_tests --suite="explorationTranslationTab" --prod_env; fi
-- if [ "$RUN_E2E_TESTS_EXTENSIONS" == 'true' ]; then travis_retry python -m scripts.run_e2e_tests --suite="extensions" --prod_env; fi
-- if [ "$RUN_E2E_TESTS_LEARNER_DASHBOARD" == 'true' ]; then travis_retry python -m scripts.run_e2e_tests --suite="learnerDashboard" --prod_env; fi
-- if [ "$RUN_E2E_TESTS_LEARNER" == 'true' ]; then travis_retry python -m scripts.run_e2e_tests --suite="learner" --prod_env; fi
-- if [ "$RUN_E2E_TESTS_SKILL_EDITOR" == 'true' ]; then travis_retry python -m scripts.run_e2e_tests --suite="skillEditor" --prod_env; fi
-- if [ "$RUN_E2E_TESTS_TOPICS_AND_SKILLS_DASHBOARD" == 'true' ]; then travis_retry python -m scripts.run_e2e_tests --suite="topicsAndSkillsDashboard" --prod_env; fi
-- if [ "$RUN_E2E_TESTS_TOPIC_AND_STORY_EDITOR" == 'true' ]; then travis_retry python -m scripts.run_e2e_tests --suite="topicAndStoryEditor" --prod_env; fi
-
-
-cache:
-  # Cache Oppia's dependencies.
-  directories:
-    - ../oppia_tools/
-    - node_modules/
-    - third_party/
-    - $HOME/.cache/pip
-    - $HOME/.cache/TravisChrome/
-    - backend_prod_files/templates/
-    - build/templates/
-    - backend_prod_files/extensions/
-    - build/extensions/
-
-before_cache:
-  # Delete python bytecode to prevent cache rebuild.
-  - find third_party -name "*.pyc" -print -delete
-  - find backend_prod_files/extensions/ -name "*.pyc" -print -delete
+- if [ "$RUN_E2E_TESTS_ADDITIONAL_EDITOR_PLAYER_FEATURES" == 'true' ]; then python -m scripts.run_e2e_tests --suite="additionalEditorFeatures" --prod_env; python -m scripts.run_e2e_tests --skip-build --skip-install --suite="additionalPlayerFeatures" --prod_env; fi
+- if [ "$RUN_E2E_TESTS_COMMUNITY_DASHBOARD" == 'true' ]; then python -m scripts.run_e2e_tests --suite="communityDashboard" --prod_env; fi
+- if [ "$RUN_E2E_TESTS_CORE_EDITOR_AND_PLAYER_FEATURES" == 'true' ]; then python -m scripts.run_e2e_tests --suite="coreEditorAndPlayerFeatures" --prod_env; fi
+- if [ "$RUN_E2E_TESTS_CREATOR_DASHBOARD" == 'true' ]; then python -m scripts.run_e2e_tests --suite="creatorDashboard" --prod_env; fi
+- if [ "$RUN_E2E_TESTS_EXPLORATION_FEEDBACK_HISTORY_TAB" == 'true' ]; then python -m scripts.run_e2e_tests --suite="explorationFeedbackTab" --prod_env; python -m scripts.run_e2e_tests --skip-build --skip-install --suite="explorationHistoryTab" --prod_env; fi
+- if [ "$RUN_E2E_TESTS_EXPLORATION_STATISTICS_TRANSLATION_TAB" == 'true' ]; then python -m scripts.run_e2e_tests --suite="explorationStatisticsTab" --prod_env; python -m scripts.run_e2e_tests --skip-build --skip-install --suite="explorationTranslationTab" --prod_env; fi
```

Comment:
If you look to the right, any suite that follows the first one has a `--skip-build` parameter, this will automatically skip the building process. That being said, because we can't build and propagate like we do for CircleCI, we have to build and install for the first test.

Reply:
Ah, I see! thanks for the info!

Should we run the one test multiple time and then only move to the next one, will that be helpful to avoid running one passed test multiple times as well as it will follow the fail-fast system?

Also, assume the first test failed and 2nd test passed then whether the build will be considered pass or fail?

AFAIK, `A; B` will eventually return the return value of B! Should we use `&&`? (In that case, the 2nd command will not run if 1st command failed, not sure whether we want that behaviour!)

Reply:
Interestingly, I've never seen a test that worked when that happened. It fails quicker like this.

Reply:
![image](https://user-images.githubusercontent.com/16653571/85215339-d64b1d00-b394-11ea-96ee-57456a4c1e0d.png)

@kevinlee12 Are you sure? Locally I see this^, do you see this in any build?